src/hotspot/cpu/x86/stubGenerator_x86_64_aes.cpp

```diff
@@ -3177,4 +3290,974 @@ void StubGenerator::aesgcm_encrypt(Register in, Register len, Register ct, Regis
+  __ vpsllq(xmm6, xmm6, 1, Assembler::AVX_128bit);
+  __ vpsrlq(xmm2, xmm2, 63, Assembler::AVX_128bit);
+  __ movdqu(xmm1, xmm2);
+  __ vpslldq(xmm2, xmm2, 8, Assembler::AVX_128bit);
+  __ vpsrldq(xmm1, xmm1, 8, Assembler::AVX_128bit);
+  __ vpor(xmm6, xmm6, xmm2, Assembler::AVX_128bit);
+
+  __ vpshufd(xmm2, xmm1, 0x24, Assembler::AVX_128bit);
+  __ vpcmpeqd(xmm2, xmm2, xmm12, Assembler::AVX_128bit);
+  __ vpand(xmm2, xmm2, xmm11, Assembler::AVX_128bit);
+  __ vpxor(xmm6, xmm6, xmm2, Assembler::AVX_128bit);
+  __ movdqu(Address(htbl, 1 * 16), xmm6); // H * 2
+  __ movdqu(xmm0, xmm6);
+  for (int i = 2; i < 9; i++) {
+      gfmul_avx2(xmm6, xmm0);
+      __ movdqu(Address(htbl, i * 16), xmm6);
+  }
+  __ ret(0);
+}
+
+void StubGenerator::ghash8_encrypt8_parallel(Register key, Register subkeyHtbl, XMMRegister ctr_blockx, XMMRegister aad_hashx,
+                                             Register in, Register out, Register ct, Register pos, bool in_order, Register rounds) {
+
+  const XMMRegister t1 = xmm0;
+  const XMMRegister t2 = xmm10;
+  const XMMRegister t3 = xmm11;
+  const XMMRegister t4 = xmm12;
+  const XMMRegister t5 = xmm13;
+  const XMMRegister t6 = xmm14;
+  const XMMRegister t7 = xmm15;
+  Label skip_reload, last_aes_rnd, aes_192, aes_256;
+
+  __ movdqu(t2, xmm1);
+  __ movdqu(Address(rsp, 16 * 0), xmm2);
+  __ movdqu(Address(rsp, 16 * 1), xmm3);
+  __ movdqu(Address(rsp, 16 * 2), xmm4);
+  __ movdqu(Address(rsp, 16 * 3), xmm5);
+  __ movdqu(Address(rsp, 16 * 4), xmm6);
+  __ movdqu(Address(rsp, 16 * 5), xmm7);
+  __ movdqu(Address(rsp, 16 * 6), xmm8);
+
+  if (in_order) {
+    __ vpaddd(xmm1, ctr_blockx, ExternalAddress(counter_mask_linc1_addr()), Assembler::AVX_128bit, rbx /*rscratch*/); //Increment counter by 1
+    __ movdqu(t5, ExternalAddress(counter_mask_linc2_addr()), rbx /*rscratch*/);
+    __ vpaddd(xmm2, ctr_blockx, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm3, xmm1, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm4, xmm2, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm5, xmm3, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm6, xmm4, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm7, xmm5, t5, Assembler::AVX_128bit);
+    __ vpaddd(xmm8, xmm6, t5, Assembler::AVX_128bit);
+    __ movdqu(ctr_blockx, xmm8);
+
+    __ movdqu(t5, ExternalAddress(counter_shuffle_mask_addr()), rbx /*rscratch*/);
+    __ vpshufb(xmm1, xmm1, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm2, xmm2, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm3, xmm3, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm4, xmm4, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm5, xmm5, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm6, xmm6, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm7, xmm7, t5, Assembler::AVX_128bit); //perform a 16Byte swap
+    __ vpshufb(xmm8, xmm8, t5, Assembler::AVX_128bit); //perform a 16Byte swap
```

Comment:
This could be written in a loop:
for (int rnum = 1; rnum <= 8; rnum++) {
    __ vpshufb(as_XMMRegister(rnum), as_XMMRegister(rnum), t5, Assembler::AVX_128bit);
}
Similar technique can be used in some places below.